src/components/FileTree/index.tsx

```diff
@@ -134,38 +139,65 @@ export const buildFileTree = (
 
 export type PublicProps = {
   onSelect: FileTreeNodeProps['onSelect'];
-  version: Version;
+  versionId: number;
 };
 
-type Props = PublicProps & ConnectedReduxProps;
+export type DefaultProps = {
+  _log: typeof log;
+};
+
+type PropsFromState = {
+  version: Version | void;
+};
+
+type Props = PublicProps & DefaultProps & PropsFromState & ConnectedReduxProps;
 
 export class FileTreeBase extends React.Component<Props> {
+  static defaultProps: DefaultProps = {
+    _log: log,
+  };
+
   renderNode = (props: TreefoldRenderPropsForFileTree) => {
     const { onSelect, version } = this.props;
 
-    return <FileTreeNode {...props} onSelect={onSelect} version={version} />;
+    if (version) {
+      return <FileTreeNode {...props} onSelect={onSelect} version={version} />;
+    }
+    // This should never happen. See the comment in mapStateToProps.
+    return null;
```

Comment:
You can return `<Loading>` here and I don't think you need a comment. We would need a state like this anyway if the component knew how to load its own data.